system/doc/design_principles/sup_princ.md

```diff
@@ -116,16 +116,74 @@ The `strategy` key is optional in this map. If it is not given, it defaults to
 
 If a child process terminates, only that process is restarted.
 
-![One_For_One Supervision](assets/sup-one_for_one.svg "One_For_One Supervision"){:
-#sup4 width=80% }
+```mermaid
+---
+title: One For One Supervision
+---
+flowchart TD
+    subgraph Legend
+        direction LR
+        t(( )) ~~~ l1[Terminated Process]
+        p(( )) ~~~ l2[Process Restarted by the Supervisor]
+    end
+
+    subgraph graph[" "]
+        s[Supervisor]
+        s --- p1((P1))
+        s --- p2((P2))
+        s --- p3((P3))
+        s --- pn((Pn))
+    end
+
+    classDef term fill:#ff8888
+    classDef restarted stroke:#00aa00,stroke-width:3px;
+    classDef legend fill-opacity:0,stroke-width:0px;
+
+    class p2,t term;
+    class p2,p restarted;
+    class l1,l2 legend;
+```
 
 ### one_for_all
 
 If a child process terminates, all other child processes are terminated, and
 then all child processes, including the terminated one, are restarted.
 
-![One_For_All Supervision](assets/sup-one_for_all.svg "One_For_All Supervision"){:
-#sup5 width=80% }
+```mermaid
+---
+title: One For All Supervision
+---
+flowchart TD
+    subgraph Legend
+        direction LR
+        t(( )) ~~~ l1[Terminated Process]
+        st(( )) ~~~ l2[Process Terminated by the Supervisor]
+        p(( )) ~~~ l3[Process Restarted by the Supervisor]
+        l4["Note:
+
+           Processes are terminated right to left
+           Processes are restarted left to right"]
+
+    end
```

Comment:
The legend here is huge, is there any way to make it smaller?

Reply:
I looked into that. I was unable to find a way to decrease the size (even after removing the note). The labels `l2` and `l3` drive the length.

I could increase the length of the node links/arrows of the supervisor graph to decrease the relative size of the legend.

Reply:
Something like this:

```mermaid
---
title: One For All Supervision
---
flowchart TD
    subgraph Legend
        direction LR
        t(( )) ~~~ l1[Terminated Process]
        st(( )) ~~~ l2[Process Terminated by the Supervisor]
        p(( )) ~~~ l3[Process Restarted by the Supervisor]
        l4["Note:

           Processes are terminated right to left
           Processes are restarted left to right"]

    end

    subgraph graph[" "]
        s[Supervisor]
        s ----- p1((P1))
        s ----- p2((P2))
        s ----- p3((P3))
        s ----- p4((P4))
        s ----- p5((P5))
        s ----- p6((P6))
        s ----- pn((Pn))
    end

    classDef term fill:#ff8888
    classDef sterm fill:#ffaa00
    classDef restarted stroke:#00aa00,stroke-width:3px;
    classDef legend fill-opacity:0,stroke-width:0px;

    class p2,t term;
    class p1,p3,p4,p5,p6,pn,st sterm;
    class p1,p2,p3,p4,p5,p6,pn,p restarted;
    class l1,l2,l3,l4 legend;
```

Reply:
Or we could increase the size of each process node:

```mermaid
---
title: One For All Supervision
---
flowchart TD
    subgraph Legend
        direction LR
        t(( )) ~~~ l1[Terminated Process]
        st(( )) ~~~ l2[Process Terminated by the Supervisor]
        p(( )) ~~~ l3[Process Restarted by the Supervisor]
        l4["Note:

           Processes are terminated right to left
           Processes are restarted left to right"]

    end

    subgraph graph[" "]
        s[Supervisor]
        s ---- p1((Process 1))
        s ---- p2((Process 2))
        s ---- p3((Process 3))
        s ---- pn((Process n))
    end

    classDef term fill:#ff8888
    classDef sterm fill:#ffaa00
    classDef restarted stroke:#00aa00,stroke-width:3px;
    classDef legend fill-opacity:0,stroke-width:0px;

    class p2,t term;
    class p1,p3,pn,st sterm;
    class p1,p2,p3,pn,p restarted;
    class l1,l2,l3,l4 legend;
```

Reply:
Ok, we can leave it as is for now. Maybe someone will come up with something better eventually.

Reply:
You should created an issue for that once this PR has been merged.